### PR TITLE
coding guidelines: partially comply with MISRA C:2012 Rule 20.7

### PR DESCRIPTION
**Pull request by Abramo-Bagnara**
MISRA C:2012 Rule 20.7 (Expressions resulting from the expansion of
macro parameters shall be enclosed in parentheses.)

Where possible, change, e.g.,

`#define EXAMPLE1(h, k) (h + k)`
 
to

`#define EXAMPLE1(h, k) ((h) + (k))`

Where this is not possible and we have a macro we cannot change, e.g.,

`#define EXAMPLE2(h, k) (h * k)`

change, e.g.,

`EXAMPLE2(x - y, a - b)`

to

`EXAMPLE2((x - y), (a - b))`

Signed-off-by: Abramo Bagnara <abramo.bagnara@bugseng.com>

### CODE REVIEW

File: tests/benchmarks/footprints/src/timer.c [hunk 1]
```diff
@@ -97,7 +97,7 @@ void run_timer(void)
 
 	k_mem_domain_add_thread(&footprint_mem_domain, tid);
 
-	k_thread_access_grant(tid, &timer0);
+	k_thread_access_grant(tid, (&timer0));
```
Comment (caspermeijn) [inline tests/benchmarks/footprints/src/timer.c:100]: I feel like this is double. The macro `k_thread_access_grant` itself is changed to have extra parenthesis, so why does this have to change as well?

File: arch/x86/include/x86_mmu.h [hunk 2]
```diff
@@ -209,7 +209,7 @@ extern pentry_t z_x86_kernel_ptables[];
 static inline pentry_t *z_x86_thread_page_tables_get(struct k_thread *thread)
 {
 #if defined(CONFIG_USERSPACE) && !defined(CONFIG_X86_COMMON_PAGE_TABLE)
-	if (!IS_ENABLED(CONFIG_X86_KPTI) ||
+	if (!(IS_ENABLED(CONFIG_X86_KPTI)) ||
```
Comment (caspermeijn) [inline arch/x86/include/x86_mmu.h:212]: Would it make sense to change the `IS_ENABLED` macro itself instead of all of its usages?

File: include/sys/cbprintf_internal.h [hunk 3]
```diff
@@ -127,9 +127,9 @@ extern "C" {
 #define Z_CBPRINTF_MUST_RUNTIME_PACKAGE(skip, ...) ({\
 	_Pragma("GCC diagnostic push") \
 	_Pragma("GCC diagnostic ignored \"-Wpointer-arith\"") \
-	int _rv = COND_CODE_0(NUM_VA_ARGS_LESS_1(__VA_ARGS__), \
-			(0), \
-			(((Z_CBPRINTF_HAS_PCHAR_ARGS(__VA_ARGS__) - skip) > 0))); \
+	bool _rv = COND_CODE_0(NUM_VA_ARGS_LESS_1(__VA_ARGS__), \
+			(false), \
```
Comment (caspermeijn) [inline include/sys/cbprintf_internal.h:131]: Is this the same coding guideline? Maybe this should be a separate commit.

**Review by caspermeijn — COMMENTED** [review]
- The first two code examples in the commit message and PR description are missing
- I think it would be better to split this up in multiple commits, where each commit fixes 1 macro

File: tests/benchmarks/footprints/src/timer.c (hunk 1 above)
Comment (Abramo-Bagnara) [reply tests/benchmarks/footprints/src/timer.c:100]: k_thread_access_grant has been changed to add parenthesis for its first argument, its definition cannot be changed to add parenthesis for its variadic arguments.

File: arch/x86/include/x86_mmu.h (hunk 2 above)
Comment (Abramo-Bagnara) [reply arch/x86/include/x86_mmu.h:212]: Unfortunately IS_ENABLED is used also in macros that need it expands only to bare 0 or 1.

**Comment by Abramo-Bagnara** [comment]
> * The first two code examples in the commit message and PR description are missing

Git has stripped lines beginning with #, I'll try to fix that.

> * I think it would be better to split this up in multiple commits, where each commit fixes 1 macro

Frankly I think that this is rather overkill and time consuming.

File: include/sys/cbprintf_internal.h (hunk 3 above)
Comment (Abramo-Bagnara) [reply include/sys/cbprintf_internal.h:131]: I'll fix that.

File: tests/benchmarks/footprints/src/timer.c (hunk 1 above)
Comment (caspermeijn) [reply tests/benchmarks/footprints/src/timer.c:100]: Ok, that makes sense.

File: arch/x86/include/x86_mmu.h (hunk 2 above)
Comment (caspermeijn) [reply arch/x86/include/x86_mmu.h:212]: Maybe there could be two macros? One for boolean and one for number. I am not sure this makes sense.

File: tests/benchmarks/footprints/src/timer.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (Abramo-Bagnara) [reply tests/benchmarks/footprints/src/timer.c:100]: The parenthesis should be around either the macro parameter names (in macro body) or the macro argument (in macro invocation). The former is preferable but sometimes this is not feasible so we are forced to the latter.

**Review by peter-mitsis — APPROVED** [review]

**Review by nashif — APPROVED** [review]

File: arch/x86/include/x86_mmu.h (hunk 2 above)
Comment (jhedberg) [reply arch/x86/include/x86_mmu.h:212]: If there are not many places that rely on the expansion to bare 0 or 1 then I think two macros should be better. It looks quite awkward to have to put parentheses around IS_ENABLED() in if-statements.
